### PR TITLE
Clean CS config, upgrade Doctrine CS to v8

### PR DESCRIPTION
**Pull request by simPod**
_Dropping PHP 7.1 is prerequisite for this https://github.com/webonyx/graphql-php/pull/470_

This includes:

- CS config cleanup
- upgrade to Doctrine CS v8
----------------
~~- upgrade to Doctrine CS v7~~

image

### CODE REVIEW

File: src/Type/Schema.php [hunk 1]
```diff
@@ -444,8 +443,11 @@ public function isPossibleType(AbstractType $abstractType, ObjectType $possibleT
             return $possibleType->implementsInterface($abstractType);
         }
 
-        /** @var UnionType $abstractType */
-        return $abstractType->isPossibleType($possibleType);
+        if ($abstractType instanceof UnionType) {
+            return $abstractType->isPossibleType($possibleType);
+        }
+
+        throw ShouldNotHappen::new();
```
[the remark this answers is not in the document]
Comment (simPod) [reply src/Type/Schema.php]: Did not notice it existed. Can do.

**Comment by vladar** [comment]
I guess we can close this PR since we had another one focused on the clean up?

**Comment by simPod** [comment]
@vladar I rebased so now it contains only CS config cleanup + slevomat CS upgrade as there are some bugs fixed.

**Comment by vladar** [comment]
Ok cool, then I guess it will stay here until we decide to drop PHP7.1 support %)

**Comment by simPod** [comment]
Tru tru

File: src/Error/Error.php [hunk 2]
```diff
@@ -207,7 +208,7 @@ public function getCategory()
         return $this->category;
     }
 
-    public function getSource() : ?Source
+    public function getSource(): ?Source
```
Comment (spawnia) [inline src/Error/Error.php:211]: I like this so much better.

File: src/Error/Error.php [hunk 3]
```diff
@@ -121,7 +122,7 @@ public function __construct(
         if ($previous instanceof ClientAware) {
             $this->isClientSafe = $previous->isClientSafe();
             $cat                = $previous->getCategory();
-            $this->category     = $cat === '' || $cat === null  ? self::CATEGORY_INTERNAL: $cat;
+            $this->category     = $cat === '' || $cat === null  ? self::CATEGORY_INTERNAL : $cat;
```
Comment (spawnia) [inline src/Error/Error.php:125]: Oh how I wish multiline ternary expressions could be automated.

```suggestion
            $this->category     = $cat === '' || $cat === null
                ? self::CATEGORY_INTERNAL
                : $cat;
```

Comment (simPod) [reply src/Error/Error.php:125]: same. But that's out of scope of this PR :P

File: src/Error/Error.php (hunk 2 above)
Comment (simPod) [reply src/Error/Error.php:211]: TBH I preferred the version before but Doctrine CS complies with PSR-12. I can live with that :)

File: phpcs.xml.dist [hunk 4]
```diff
@@ -13,90 +13,37 @@
...
 
     <rule ref="Doctrine">
-        <!-- Disable PHP7+ features that might cause BC breaks for now -->
-        <exclude name="SlevomatCodingStandard.Classes.ClassConstantVisibility.MissingConstantVisibility" />
-        <exclude name="SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingParameterTypeHint" />
-        <exclude name="SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingReturnTypeHint" />
-        <!-- Enable when Slevomat starts supporting variadics, see https://github.com/slevomat/coding-standard/issues/251 -->
-        <exclude name="SlevomatCodingStandard.Namespaces.UnusedUses.UnusedUse" />
-    </rule>
+        <!-- Remove when PHP 7.4 is minimum version requirement -->
```
Comment (spawnia) [inline phpcs.xml.dist:16]: It is time.

**Review by spawnia — CHANGES_REQUESTED** [review]

**Review by spawnia — APPROVED** [review]
Cool

**Pull request merged**